contracts/exchange/contracts/src/MixinExchangeCore.sol

```diff
@@ -271,21 +280,126 @@ contract MixinExchangeCore is
 
         // Emit a Fill() event THE HARD WAY to avoid a stack overflow.
         // All this logic is equivalent to:
-        emit Fill(
-            order.makerAddress,
-            order.feeRecipientAddress,
-            order.makerAssetData,
-            order.takerAssetData,
-            order.makerFeeAssetData,
-            order.takerFeeAssetData,
-            fillResults.makerAssetFilledAmount,
-            fillResults.takerAssetFilledAmount,
-            fillResults.makerFeePaid,
-            fillResults.takerFeePaid,
-            takerAddress,
-            msg.sender,
-            orderHash
+        // emit Fill(
+        //     order.makerAddress,
+        //     order.feeRecipientAddress,
+        //     orderHash,
+        //     takerAddress,
+        //     msg.sender,
+        //     fillResults.makerAssetFilledAmount,
+        //     fillResults.takerAssetFilledAmount,
+        //     fillResults.makerFeePaid,
+        //     fillResults.takerFeePaid,
+        //     fillResults.protocolFeePaid,
+        //     msg.value >= fillResults.protocolFeePaid,
+        //     order.makerAssetData,
+        //     order.takerAssetData,
+        //     order.makerFeeAssetData,
+        //     order.takerFeeAssetData
+        // );
+
+        // There are 15 total fields in the `Fill()` event. Out of all of these fields,
+        // 3 are indexed fields. Due to the evm semantics of event logging, the 12 non-indexed
+        // fields will be abi-encoded into a bytes structure and then put into the `data` slot
+        // of the event logs.
+        //
+        // Since there are 12 fields, we will need 12 * 32 = 384 bytes to store the fields. Due
+        // to the fact that 4 of the fields are of type `bytes`, there is added complexity in that
+        // these fields in the log data are actually byte offsets to the length-prefaced `bytes`
+        // of the field. This means that we also need to account for 4 length slots -- 4 * 32 = 128
+        // more bytes in the log data -- and for the length of the bytes themselves -- total_length =
+        // order.makerAssetData.length + order.takerAssetData.length + order.makerFeeAssetData.length
+        // + order.takerFeeAssetData.length.
+        //
+        // Altogether, the log data will be 384 + 128 + total_length = 512 + total_length bytes long,
+        // which explains the memory allocation of the bytes below:
+        bytes memory logData = new bytes(
```

Comment:
Do we have a sense of the gas differences using this approach vs logging 2 separate events?